### PR TITLE
Allowing monitoring interval to be configurable per monitor

### PR DESCRIPTION
**Pull request by itzg**
# Resolves

https://jira.rax.io/browse/SALUS-592

# What

Instead of hard coding the overall agent collection interval for telegraf these changes
- use a locally configurable default
- accept a monitoring interval from the incoming config instruction to override that default interval per monitor

## How to test

Updated unit tests

### CODE REVIEW

File: telemetry_edge/telemetry-edge.pb.go [hunk 1]
```diff
@@ -1146,77 +1155,78 @@ func init() {
...
-	0x8b, 0xf6, 0x3f, 0x06, 0x6c, 0xf4, 0xb2, 0x1f, 0xba, 0xce, 0xf0, 0x8a, 0x70, 0x4e, 0xfa, 0x11,
-	0x43, 0x2f, 0xa1, 0xd1, 0x11, 0x82, 0x78, 0x37, 0xaa, 0x04, 0xe8, 0xd1, 0xcc, 0xdf, 0x56, 0x6b,
-	0x7d, 0xe1, 0xab, 0x6e, 0x15, 0x5e, 0x14, 0xd1, 0x2b, 0xa8, 0x4f, 0x66, 0x1d, 0x5a, 0xc7, 0xf3,
-	0xb3, 0xb0, 0x85, 0xf0, 0xe2, 0x28, 0x2c, 0x20, 0x3b, 0x1d, 0x4e, 0x93, 0x81, 0x58, 0xc7, 0xd9,
-	0xb2, 0xf5, 0x04, 0x2f, 0x1d, 0x67, 0x92, 0x03, 0x53, 0xc9, 0xa0, 0x47, 0x38, 0x3f, 0xdd, 0x5a,
-	0x1b, 0x78, 0x89, 0x9c, 0x0a, 0xe8, 0x1d, 0x6c, 0x2d, 0xcf, 0x1a, 0xda, 0xc0, 0x8b, 0xce, 0xd6,
-	0x53, 0x7c, 0x7f, 0x8e, 0xad, 0xc2, 0xe1, 0x2e, 0xec, 0x78, 0xd1, 0x10, 0x33, 0xe2, 0xdd, 0xf2,
-	0x98, 0x78, 0x14, 0x73, 0x12, 0x24, 0x1c, 0x73, 0xca, 0x46, 0xbe, 0x47, 0xf9, 0x55, 0x55, 0xfd,
-	0x1b, 0xbf, 0xfc, 0x37, 0x00, 0x00, 0xff, 0xff, 0xee, 0x02, 0xd0, 0x03, 0x33, 0x0b, 0x00, 0x00,
+	// 1135 bytes of a gzipped FileDescriptorProto
```
Comment (harageth) [inline telemetry_edge/telemetry-edge.pb.go]: Why is this a hardcoded byte array?

Comment (itzg) [reply telemetry_edge/telemetry-edge.pb.go]: This file is generated by protoc. I highly recommend you don't look at it :) It used to be generated at build time, but it was too complicated to make the GCB build download protoc, etc :(

File: telemetry_edge/telemetry-edge.pb.go [hunk 2]
```diff
@@ -508,10 +508,12 @@ type ConfigurationOp struct {
 	Type     ConfigurationOp_Type `protobuf:"varint,3,opt,name=type,proto3,enum=ConfigurationOp_Type" json:"type,omitempty"`
```
Comment (itzg) [inline telemetry_edge/telemetry-edge.pb.go]: I should have pointed out that this file is not suitable for human consumption. It is generated by using this goal in the Makefile

https://github.com/racker/salus-telemetry-envoy/blob/4eef46acea608ffc019d37d2172f9c35af3294b9/Makefile#L27-L28

File: telemetry_edge/telemetry-edge.pb.go (hunk 1 above)
Comment (harageth) [reply telemetry_edge/telemetry-edge.pb.go]: Ahhhh that last part was the information I was looking for. I guess the next question is why is the file being stored like this and not as its own file that gets imported?

Comment (itzg) [reply telemetry_edge/telemetry-edge.pb.go]: I don't follow your question. Are you asking about how we're organizing this in our repo or why the protoc generator embedded the binary file content like it did?

Comment (harageth) [reply telemetry_edge/telemetry-edge.pb.go]: Ohhhhh I think your other comment makes it clear that this file is whats being generated. I hadn't understood the other comment till now.

Comment (itzg) [reply telemetry_edge/telemetry-edge.pb.go]: Yeah, in short: don't look at this file :)

To finish the other thought though, unlike Java jars (which are just zip files) Go doesn't have a native way to embed static content into the final build:

https://stackoverflow.com/questions/13904441/whats-the-best-way-to-bundle-static-resources-in-a-go-program

**Pull request merged**